### PR TITLE
Update TSApi for TShock v4.3.26

### PR DESCRIPTION
**Pull request by QuiCM**
* Adds a patch for 0-length crash
* updates OTAPI version
* updates JSON version

Merges gen-dev -> master

### CODE REVIEW

File: TShock.Modifications.Bootstrapper/Program.cs [hunk 1]
```diff
@@ -15,15 +15,30 @@ static void Main(string[] args)
 			string sourceAsm = null;
 			string modificationGlob = null;
 			string outputPath = null;
+			string folder = null;
```
Comment (Ijwu) [inline TShock.Modifications.Bootstrapper/Program.cs:18]: Shouldn't this be defaulted to something?

**Review by Ijwu — APPROVED** [review]

File: TShock.Modifications.Bootstrapper/Program.cs (hunk 1 above)
Comment (Ijwu) [reply TShock.Modifications.Bootstrapper/Program.cs:18]: Only the one question, otherwise lgtm

Comment (QuiCM) [reply TShock.Modifications.Bootstrapper/Program.cs:18]: I have no idea

Comment (hakusaro) [reply TShock.Modifications.Bootstrapper/Program.cs:18]: I agree

Comment (hakusaro) [reply TShock.Modifications.Bootstrapper/Program.cs:18]: (w.r.t. "I have no idea")

**Review by hakusaro — APPROVED** [review]
lgtm_6

File: TShock.Modifications.Bootstrapper/Program.cs (hunk 1 above)
Comment (Ijwu) [reply TShock.Modifications.Bootstrapper/Program.cs:18]: It seems to have been defaulted to "Debug" before the change. Should that be the default value?

Comment (QuiCM) [reply TShock.Modifications.Bootstrapper/Program.cs:18]: The csproj has a build task that generates env.config with the current configuration set:
https://github.com/Pryaxis/TerrariaAPI-Server/blob/1f1901ac9dc891cd4663585a2b61a6179726407f/TShock.Modifications.Bootstrapper/TShock.Modifications.Bootstrapper.csproj#L100-L104
`folder` should refer to the current configuration when building via anything that uses the csproj

**Pull request merged**